app/models/spree_avatax/sales_invoice.rb

```diff
@@ -25,6 +25,11 @@ def generate(order)
 
       return if order.completed? || !SpreeAvatax::Shared.taxable_order?(order)
 
+      taxable_records = order.line_items + order.shipments
+      taxable_records.each do |taxable_record|
+        taxable_record.update_column(:pre_tax_amount, taxable_record.discounted_amount.round(2))
+      end
```

Comment:
@gmacdougall this was already happening [later on](https://github.com/solidusio/solidus_avatax/blob/master/app/models/spree_avatax/sales_shared.rb#L33), this is just to ensure that the values are still set in case the invoice cannot be generated for any reason (e.g. Avatax network timeout). Storing the pre_tax_amount is even [overriden](https://github.com/solidusio/solidus_avatax/blob/master/app/models/spree/tax_rate_decorator.rb#L15) given we handle it at a different time as the comment mentions.